src/test/java/io/asyncer/r2dbc/mysql/SslTunnelIntegrationTest.java

```diff
@@ -0,0 +1,297 @@
+        final SslContext sslContext = SslContextBuilder.forServer(server.key(), server.cert()).build();
+        sslTunnelServer = new SslTunnelServer("localhost", 3306, sslContext);
+        sslTunnelServer.setUp();
+    }
+
+    @After
+    public void tearDown() throws InterruptedException {
+        server.delete();
+        client.delete();
+        sslTunnelServer.tearDown();
+    }
+
+    @Test
+    public void sslTunnelConnectionTest() {
+        final String password = System.getProperty("test.mysql.password");
+        assertThat(password).withFailMessage("Property test.mysql.password must exists and not be empty")
+                            .isNotNull()
+                            .isNotEmpty();
+
+        final MySqlConnectionConfiguration configuration = MySqlConnectionConfiguration
+                .builder()
+                .host("localhost")
+                .port(sslTunnelServer.getLocalPort())
+                .connectTimeout(Duration.ofSeconds(3))
+                .user("root")
+                .password(password)
+                .database("r2dbc")
+                .createDatabaseIfNotExist(true)
+                .sslMode(SslMode.TUNNEL)
+                .sslKey(client.privateKey().getAbsolutePath())
+                .sslCert(client.certificate().getAbsolutePath())
+                .sslCa(server.certificate().getAbsolutePath())
+                .build();
+
+        final MySqlConnectionFactory connectionFactory = MySqlConnectionFactory.from(configuration);
+
+        final MySqlConnection connection = connectionFactory.create().block();
+        connection.createStatement("SELECT 3").execute()
+                  .flatMap(it -> it.map((row, rowMetadata) -> row.get(0)))
+                  .doOnNext(it -> assertThat(it).isEqualTo(3L))
+                  .blockLast();
+
+        connection.close().block();
+    }
+
+    private static class SslTunnelServer {
+
+        private final String remoteHost;
+
+        private final int remotePort;
+
+        private final SslContext sslContext;
+
+        private volatile ChannelFuture channelFuture;
+
+
+        private SslTunnelServer(String remoteHost, int remotePort, SslContext sslContext) {
+            this.remoteHost = remoteHost;
+            this.remotePort = remotePort;
+            this.sslContext = sslContext;
+        }
+
+        void setUp() {
+            // Configure the server.
+            try {
+                ServerBootstrap b = new ServerBootstrap();
+                b.localAddress(0)
+                 .group(new NioEventLoopGroup())
+                 .channel(NioServerSocketChannel.class)
+                 .childHandler(new ProxyInitializer(remoteHost, remotePort, sslContext))
+                 .childOption(ChannelOption.AUTO_READ, false);
+
+
+                // Start the server.
+                channelFuture = b.bind().sync();
+
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
```

Comment:
The exception can be thrown directly, and `SslTunnelIntegrationTest.setUp()` can be `throws CertificateException, SSLException, InterruptedException`. 

```suggestion
        void setUp() throws InterruptedException {
            // Configure the server.
            ServerBootstrap b = new ServerBootstrap();
            b.localAddress(0)
                .group(new NioEventLoopGroup())
                .channel(NioServerSocketChannel.class)
                .childHandler(new ProxyInitializer(remoteHost, remotePort, sslContext))
                .childOption(ChannelOption.AUTO_READ, false);

            // Start the server.
            channelFuture = b.bind().sync();
        }
```